weatherAplication/Modules/Home/View/HomeView.swift

```diff
@@ -0,0 +1,101 @@
+import SnapKit
+import UIKit
+
+class HomeView: UIView {
+    private let kLogoWidth: CGFloat = 250
+    private let kLogoHeight: CGFloat = 250
+    private let kSideMargin: CGFloat = 10
+    private let kTopMargin: CGFloat = 20
+    private let kBottomMargin: CGFloat = 15
+    private let kTableBottomMargin: CGFloat = 50
+    private let aplicationImage = R.image.weather_symbol()
+    private let logoImageView = UIImageView()
+    private let aplicationNameLabel = UILabel()
+    internal let searchButton = SearchButton()
+    private let aplicationNameString: String = "MyWeather"
+    internal let lastCityNamesTableView = UITableView()
+
+    private var viewModel: HomeViewModelContract
+
+    init(viewModelContract: HomeViewModelContract) {
+        self.viewModel = viewModelContract
+        super.init(frame: .zero)
+        setupView()
+        setupConstraints()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func setupView() {
+        self.backgroundColor = .blue
+        logoImageView.image = aplicationImage
+        self.addSubview(logoImageView)
+        aplicationNameLabelSetup()
+        lastCityNamesTableViewSetup()
+        self.addSubview(searchButton)
+    }
+
+    func aplicationNameLabelSetup() {
+        aplicationNameLabel.text = aplicationNameString
+        aplicationNameLabel.textColor = .white
+        aplicationNameLabel.textAlignment = .center
+        aplicationNameLabel.backgroundColor = .black
+        self.addSubview(aplicationNameLabel)
+    }
+
+    func lastCityNamesTableViewSetup() {
+        lastCityNamesTableView.dataSource = self
+        lastCityNamesTableView.delegate = self
+        lastCityNamesTableView.backgroundColor = .cyan
+        self.addSubview(lastCityNamesTableView)
+        lastCityNamesTableView.register(HomeTableViewCell.self, forCellReuseIdentifier: "cell")
+    }
+
+    func setupConstraints() {
+
+        logoImageView.snp.makeConstraints { make in
+            make.width.equalTo(kLogoWidth)
+            make.height.equalTo(kLogoHeight)
+            make.top.equalTo(self.snp.topMargin).offset(kTopMargin)
```

Comment:
to powinno sie dac zapisac przez equalToSuperview().offset(), chyba ze czegos nie widze

Reply:
w niektorych z constraintow w tej klasie podobnie